resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -250,11 +250,32 @@ private[spark] object Config extends Logging {
       .stringConf
       .createOptional
 
+  // the definition of a label in DNS (RFC 1123).
+  private val dns1123LabelFmt = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+
+  private val podConfValidator = (s"^$dns1123LabelFmt(\\.$dns1123LabelFmt)*$$").r.pattern
+
+  // The possible longest executor name would be "$prefix-exec-${Long.MaxValue}"
```

Comment:
ditto. `${Long.MaxValue}` is correct?

Reply:
https://github.com/apache/spark/blob/465a2503ac30e48e1cb8a68d181b92bd90d530fd/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala#L49

I guess Int is correct as we use int type with other cluster managers. Shall we fix this line too in this PR?

Reply:
It looks worth to have a separate PR, @yaooqinn . Could you make a new PR for `AtomicLong(0L)` change?

Reply:
https://github.com/apache/spark/pull/32837 I have sent a PR for this